### PR TITLE
PWN-3748 - Swap/Send screen selection mechanism / Part 2

### PR DESCRIPTION
**Pull request by 1sakov**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-3100
https://p2pvalidator.atlassian.net/browse/PWN-3748

### CODE REVIEW

**Review by davran1001 — APPROVED** [review]

File: app/src/main/res/layout/widget_account_fee_view.xml [hunk 1]
```diff
@@ -41,7 +41,7 @@
         app:layout_constraintEnd_toStartOf="@id/arrowImageView"
         app:layout_constraintStart_toEndOf="@id/accountImageView"
         app:layout_constraintTop_toBottomOf="@id/accountFeeTextView"
-        tools:text="0.509 USDC" />
+        tools:text="0.509 USDC рЙ\nЙЙЙЙЙЙЙррЙЙЙ\nЙЙЙЙЙЙЙЙЙЙЙЙ" />
```
Comment (kuznetsov) [inline app/src/main/res/layout/widget_account_fee_view.xml:44]: Redundant text from design grooming)

**Review by kuznetsov — APPROVED** [review]

**Comment by 1sakov** [comment]
don't review guys, please, yet

**Review by kabramovp2p — APPROVED** [review]

**Review by gslevinkov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/send/ui/main/SendPresenter.kt [hunk 2]
```diff
@@ -89,46 +90,34 @@ class SendPresenter(
...
                 val userPublicKey = tokenKeyProvider.publicKey
-                solToken = userTokens.find { it.isSOL && it.publicKey == userPublicKey }
-                    ?: throw IllegalStateException("No SOL account found")
-                token = initialToken ?: solToken
+                state.solToken = userInteractor.getUserTokens().find { it.isSOL && it.publicKey == userPublicKey }
+                token = state.initialToken ?: state.solToken
 
-                calculateTotal(sendFee = null)
-                view?.showNetworkDestination(networkType)
+                sendInteractor.initialize(state.solToken!!)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/send/ui/main/SendPresenter.kt:116]: i can suggest extracting `userInteractor.getUserTokens().find { it.isSOL && it.publicKey == userPublicKey }` to a local val and using it in both places to remove !!

File: app/src/main/java/org/p2p/wallet/send/ui/main/SendPresenter.kt [hunk 3]
```diff
@@ -89,46 +90,34 @@ class SendPresenter(
...
 
-                val userTokens = userInteractor.getUserTokens()
+                /*
+                 * We should find SOL anyway because SOL is needed for Selection Mechanism
+                 * */
                 val userPublicKey = tokenKeyProvider.publicKey
-                solToken = userTokens.find { it.isSOL && it.publicKey == userPublicKey }
-                    ?: throw IllegalStateException("No SOL account found")
-                token = initialToken ?: solToken
+                state.solToken = userInteractor.getUserTokens().find { it.isSOL && it.publicKey == userPublicKey }
```
[the remark this answers is not in the document]
Comment (1sakov) [reply app/src/main/java/org/p2p/wallet/send/ui/main/SendPresenter.kt:113]: That's temporary implementation. It was too hard to control the states, I just needed all the variables in one object. 
Then we could make some state  updates, now it's just a container

File: app/src/main/java/org/p2p/wallet/swap/ui/orca/OrcaSwapPresenter.kt [hunk 4]
```diff
@@ -100,8 +100,13 @@ class OrcaSwapPresenter(
         launch {
             view?.showFullScreenLoading(true)
             try {
+                /*
+                * We should find SOL anyway because SOL is needed for Selection Mechanism
+                * */
+                solToken = userInteractor.getUserTokens().find { it.isSOL }
```
[the remark this answers is not in the document]
Comment (1sakov) [reply app/src/main/java/org/p2p/wallet/swap/ui/orca/OrcaSwapPresenter.kt:106]: agree

**Pull request merged**